src/components/SplitField/SplitField.jsx

```diff
@@ -1,45 +1,70 @@
-import { forwardRef } from 'react';
-import { Button, Card, FormControl, InputGroup } from 'react-bootstrap';
-import { faTrashCan } from '@fortawesome/free-regular-svg-icons';
-import { faPlusMinus } from '@fortawesome/free-solid-svg-icons';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { EXACT_AMOUNTS, PERCENTAGES, SHARES } from '../../constants';
+import { forwardRef, useState, useContext } from "react";
+import { Button, Card, FormControl, InputGroup } from "react-bootstrap";
+import { faTrashCan } from "@fortawesome/free-regular-svg-icons";
+import { faPlusMinus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { EQUALLY, EXACT_AMOUNTS, PERCENTAGES, SHARES } from "../../constants";
+import SplitDataContext from "../../store/split-data-context";
 
-import './styles.scss';
+import "./styles.scss";
 
 export const SplitField = forwardRef((props, ref) => {
+  const splitState = useContext(SplitDataContext);
+
+  const [exactAmtInput, setExactAmtInput] = useState("");
+
   const withAdjustmentModifier =
-    props.splitType === 'Shares' ? 'split-field--with-adjustment' : '';
+    props.splitType === "Shares" ? "split-field--with-adjustment" : "";
 
   const handleRemoveClick = () => {
     props.onRemove();
   };
 
+  const handleChange = (e) => {
+    const inputId = props.id;
+    console.log(inputId);
+    splitState.handleSplitFieldInputChange(e.target.value, inputId);
+  };
+
+  const handleNameChange = (e) => {
+    const inputId = props.id;
+    splitState.handleNameChange(e.target.value, inputId);
+  };
+
   return (
     <div className={`split-field ${withAdjustmentModifier}`} ref={ref}>
       <div className="split-field__index">{props.position}</div>
 
-      <FormControl placeholder="Name" className="split-field__name" />
+      <FormControl
+        placeholder="Name"
+        className="split-field__name"
+        onChange={handleNameChange}
+        value={props.name}
+      />
 
       {[EXACT_AMOUNTS, PERCENTAGES, SHARES].includes(props.splitType) && (
```

Comment:
You probably want to grab `splitType` from `splitState` since you're not using `props` anymore
```suggestion
      {[EXACT_AMOUNTS, PERCENTAGES, SHARES].includes(splitState.splitType) && (
```